### ISSUE TITLE
[x/programs] add a macros for generating public methods

### ISSUE THREAD

**Issue by iFrostizz** [ava-labs/hypersdk#943]
We find ourselves writing a lot of boilerplate code to write public getters of state keys.
This is the case of many methods of the upcoming ERC20 token example. https://github.com/ava-labs/hypersdk/blob/b646347d25954e93bf19368862a47ae985a3a100/x/programs/test/programs/simple_token/src/lib.rs
It would be great to have a macro attribute to be placed on the relevant `StateKeys` variants to generate this code for us.

**Comment by richardpringle** [comment]
Yeah, I think this is a good idea for now. I think there might be better ways to do this at the VM level, but until we've plugged programs into an actual VM, this is probably a good stop-gap

**Comment by richardpringle** [comment]
Sorry, let me rephrase: I think *using* the `state_keys` macro to create getters is a good idea. There are multiple ways we can accomplish this and I'm not sure that annotating every variant is the best way to accomplish this.
